src/SFA.DAS.EmployerIncentives.Domain/ApprenticeshipIncentives/ApprenticeshipIncentive.cs

```diff
@@ -93,6 +94,36 @@ public void CalculateEarnings(CollectionCalendar collectionCalendar)
             Model.RefreshedLearnerForEarnings = false;
         }
 
+        public void RecalculateEarnings(CollectionCalendar collectionCalendar)
+        {
+            if (Model.Status == IncentiveStatus.Withdrawn || Model.Status == IncentiveStatus.Stopped)
+            {
+                return;
+            }
+
+            var recalculationRequired = false;
+            foreach(var pendingPayment in PendingPayments)
+            {
+                if (!ExistingPendingPaymentHasBeenPaid(pendingPayment))
+                {
+                    var existingPendingPaymentModel = pendingPayment.GetModel();
+                    Model.PendingPaymentModels.Remove(existingPendingPaymentModel);
```

Comment:
No pending payment deleted event is thrown.

Reply:
That's been added